lib/modules/platform/gitlab/index.ts

```diff
@@ -970,12 +971,50 @@ export async function addAssignees(
     for (const assignee of assignees) {
       assigneeIds.push(await getUserID(assignee));
     }
-    const url = `projects/${
+    let url = `projects/${
       config.repository
     }/merge_requests/${iid}?${getQueryString({
       'assignee_ids[]': assigneeIds,
     })}`;
-    await gitlabApi.putJson(url);
+    const mr = await gitlabApi.putJson<GitLabMergeRequest>(url);
+
+    // GitLab API returns empty array of assignees in case some of them are not project members
+    if (mr.body.assignees?.length === 0) {
+      if (assigneeIds.length === 1) {
+        logger.warn('Failed to add single assignee');
+        return;
+      } else if (assigneeIds.length > 1) {
```

Comment:
we should short circuit before if assignees is empty so this is always true

Reply:
I'm not sure I understand what you mean. We can not just break if the response from gitlab API on assignees is 0 since that is the bugged case. If you mean empty assignees on user input, it is already taken care of before invoking the addAssignees method [here](https://github.com/renovatebot/renovate/blob/f04a1e32a20889c2c00be9e0aa54e5a1e7e1b26b/lib/workers/repository/update/pr/participants.ts#L46)